crypto/src/mls/credential.rs

```diff
@@ -23,84 +26,49 @@ pub struct CertificateBundle {
     pub private_key: SignaturePrivateKey,
 }
 
+impl CertificateBundle {
+    /// Reads the client_id from the leaf certificate
+    pub fn get_client_id(&self) -> CryptoResult<ClientId> {
+        let leaf = self.certificate_chain.get(0).ok_or(CryptoError::InvalidIdentity)?;
+        let identity = leaf.extract_identity().map_err(|_| CryptoError::InvalidIdentity)?;
+        Ok(identity.client_id.as_bytes().into())
+    }
+}
+
 #[cfg(test)]
 impl CertificateBundle {
     /// Basic credentials are generated once clients are created
     /// It will effectively return `None`
     pub fn rand_basic() -> CredentialSupplier {
-        |_: MlsCiphersuite| None
+        |_: ClientId, _: MlsCiphersuite| None
     }
 
     /// Generates a supplier that is later turned into a [openmls::prelude::CredentialBundle]
-    pub fn rand_certificate_bundle() -> CredentialSupplier {
-        |cs: MlsCiphersuite| {
-            // generate the leaf certificate
-            let (leaf_kp, leaf_sk) = Self::key_pair(cs);
-            let leaf_params = Self::certificate_params(leaf_kp, false);
-            Some(Self::new_certificate_bundle_from_leaf(cs, leaf_params, leaf_sk))
-        }
+    pub fn rand_certificate_bundle_old() -> CredentialSupplier {
+        todo!()
```

Comment:
Careful with `todo!()`; It panics when ran so maybe we want to return a dummy `CredentialSupplier` containing descriptive labels that it's not implemented and contains a dummy value?

Reply:
Oops, it's an omission. Thanks for catching.